src/openzaak/utils/middleware.py

```diff
@@ -99,3 +102,53 @@ def process_view(self, request, view_func, view_args, view_kwargs):
         if not disabled:
             return None
         return HttpResponseNotFound()
+
+
+# See https://github.com/Geonovum/KP-APIs/blob/master/Werkgroep%20API%20strategie/extensies/ext-versionering.md
+class Warning:
+    def __init__(self, code: int, agent: str, text: str):
+        self.code = code
+        self.agent = agent
+        self.text = text
+
+    def __str__(self):
+        return f'{self.code} "{self.agent}" "{self.text}"'
+
+
+class DeprecationMiddleware:
+    """
+    Include a header outputting a deprecation warning.
+    """
+
+    def __init__(self, get_response=None):
+        self.get_response = get_response
+
+    def __call__(self, request):
+        if self.get_response is None:
+            return None
+
+        response = self.get_response(request)
+
+        warning = getattr(request, "_warning", None)
+        if warning:
+            response[WARNING_HEADER] = str(warning)
+
+        return response
+
+    def process_view(self, request, callback, callback_args, callback_kwargs):
```

Comment:
Checked and `process_view` is still supported in Django 3.2 (https://docs.djangoproject.com/en/3.2/topics/http/middleware/#process-view)